module_vd_gas_wesely.f90

```diff
@@ -0,0 +1,211 @@
+        !    deltaz     Layer 1 midpoint height(where the Vd is evaluated)           (unit = m)
+        !    psih       similarity stability correction term                         (unit = 1)
+        !    ustar      friction velocity                                            (unit = m/s)
+        !    diffrat    ratio of molecular diffusivity of water to species           (unit = 1)
+        !    henry      Henry's Law constant                                         (unit = M/atm)
+        !    henso2     Henry's Law constant for SO2                                 (unit = M/atm)
+        !    f0         normalized reactivity parameter                              (unit = 1)
+        !    rscale     user-defined surface resistance scalling factor              (unit = 1)
+        !    ts         surface temperature                                          (unit = C NOT K)
+        !    dstress    adjustment factors for drought stress                        (unit = matrix)
+        !    solflux    Solar radiation flux                                         (unit = W/m^2)
+        !    ri         baseline minimum stomatal resistance                         (unit = s/m)
+        !    rlu        baseline upper canopy (cuticle) resistance                   (unit = s/m)
+        !    rac        baseline canopy height/density resistance                    (unit = s/m)
+        !    rlcs       baseline SO2 lower canopy resistance                         (unit = s/m)
+        !    rlco       baseline O3 lower canopy resistance                          (unit = s/m)
+        !    rgss       baseline SO2 ground surface resistance                       (unit = s/m)
+        !    rgso       baseline O3 ground surface resistance                        (unit = s/m)
+        !    icdocn     flag indicating an over-ocean cell                           (unit = 1)
+
+        !Output arguments:
+        !    vd         deposition velocity                                          (unit = m/s)
+
+        !Routines called:
+        !    none
+
+        !Called by:
+        !    should be part of dry deposition code
+
+        !Step 0. Define variables type & constant values
+        integer, intent(in) :: ilu 
+        integer, intent(in) :: istress
+        integer, intent(in) :: iwet
+        integer, intent(in) :: iso2
+        integer, intent(in) :: io3
+        integer, intent(in) :: icdocn
+        real, intent(in)    :: z0
+        real, intent(in)    :: deltaz
+        real, intent(in)    :: psih
+        real, intent(in)    :: ustar
+        real, intent(in)    :: diffrat
+        real, intent(in)    :: henry
+        real, intent(in)    :: henso2
+        real, intent(in)    :: f0
+        real, intent(in)    :: rscale
+        real, intent(in)    :: ts
+        real, intent(in)    :: dstress(0:5)
+        real*8, intent(in)  :: solflux ! why does this one have precision but the others dont? 
```

Comment:
I just copy from CAMx code, they have a double precision for solflux. 
Also, in CAMx code, they only define solflux and diffrat here. I added all other input and output parameter types here.